pkg/server/backgroundsvcs/background_services.go

```diff
@@ -152,6 +173,10 @@ func (r *BackgroundServiceRegistry) start(ctx context.Context) error {
 }
 
 func (r *BackgroundServiceRegistry) run(ctx context.Context) error {
+	if err := r.start(ctx); err != nil {
```

Comment:
would it work to add it to `NewBasicService` above?

Reply:
looks like it does! changed